iceoryx_posh/source/popo/ports/server_port_roudi.cpp

```diff
@@ -38,19 +39,108 @@ ServerPortRouDi::MemberType_t* ServerPortRouDi::getMembers() noexcept
     return reinterpret_cast<MemberType_t*>(BasePort::getMembers());
 }
 
+ConsumerTooSlowPolicy ServerPortRouDi::getClientTooSlowPolicy() const noexcept
+{
+    return static_cast<ConsumerTooSlowPolicy>(getMembers()->m_chunkSenderData.m_subscriberTooSlowPolicy);
+}
+
 cxx::optional<capro::CaproMessage> ServerPortRouDi::tryGetCaProMessage() noexcept
 {
+    // get offer state request from user side
+    const auto offeringRequested = getMembers()->m_offeringRequested.load(std::memory_order_relaxed);
+
+    const auto isOffered = getMembers()->m_offered.load(std::memory_order_relaxed);
+
+    if (isOffered)
```

Comment:
A personal note, I would prefer a structure like
```cpp
if ( isOffered && !offeringRequested ) {
  capro::CaproMessage caproMessage(capro::CaproMessageType::STOP_OFFER, this->getCaProServiceDescription());
  return dispatchCaProMessageAndGetPossibleResponse(caproMessage);
} else if ( !isOffered && offeringRequested ) {
  capro::CaproMessage caproMessage(capro::CaproMessageType::OFFER, this->getCaProServiceDescription());
  return dispatchCaProMessageAndGetPossibleResponse(caproMessage);
}
```

Which is for me much more readable. Otherwise I have two additional if statements and my brain has to fiddle this whole thing together. I think with the approach above you cut the cyclomatic complexity in half.

Reply:
I disagree on this. It is not easily visible when the else branch is taken since it does not only depend on one "complex" condition on the `else if` but one has also to parse the `if` condition and suddenly there is a 2x2 matrix which decides if the branch is taken. Whit the current code, one can easily see that the `else` branch is only taken when not offered and this matches the logic from `dispatchCaProMessageAndGetPossibleResponse` ... let's wait for the opinion of the second reviewer. There is a good chance to get a third opinion.

Reply:
@elfenpiff ping :)